GSoC-2025/DataShuttle:Cross-platform Executable Package (Aakash Kharb).md

```diff
@@ -0,0 +1,94 @@
+# DataShuttle: Cross-platform Executable Package (Aakash Kharb)
+
+## Personal Details
+
+- **Full name**: Aakash Kharb
+- **Email**: [akharbrtk2@gmail.com](mailto:akharbrtk2@gmail.com)
+- **GitHub username**: aakash-test7
+- **Zulip username**: Aakash Kharb
+- **Linkedin**: [https://linkedin.com/in/aakash-kharb/](https://linkedin.com/in/aakash-kharb/)
+- **Location & Time-zone**: India, Indian Standard Time (GMT+5:30)
+
+## Code Contribution
+
+- [https://github.com/neuroinformatics-unit/datashuttle/pull/473](https://github.com/neuroinformatics-unit/datashuttle/pull/473)
+- [https://github.com/neuroinformatics-unit/datashuttle/pull/471](https://github.com/neuroinformatics-unit/datashuttle/pull/471)
+
+## Project Proposal
+
+### Synopsis
+
+DataShuttle currently requires Conda for installation, which limits accessibility for non-technical users. This project aims to create a standalone, cross-platform executable package for DataShuttle that works on Windows, macOS, and Linux without requiring Conda or Python installation. The solution will address Textual's rendering limitations in native terminals by potentially bundling a compatible terminal environment.
+
+### Deliverables
+
+- Research report on packaging approaches for Textual applications
+- Implemented solution for cross-platform DataShuttle distribution
+- CI/CD integration for automated builds
+- Comprehensive documentation for users and maintainers
+- Testing framework for packaged versions
+
+### Implementation Timeline
+
+| Period          | Tasks                                                         | Hours/Week |
+|-----------------|---------------------------------------------------------------|------------|
+| Community Bonding | Familiarize with codebase, setup dev environment, finalize research plan | 20         |
+| Week 1-2        | Research packaging solutions (PyInstaller, PyOxidizer, Briefcase), evaluate approaches | 35         |
+| Week 3-5        | Prototype selected solution, address terminal rendering issues | 40         |
+| Week 6-7        | Implement CI/CD integration, automated builds                 | 40         |
+| Week 8-9        | Comprehensive testing across platforms                        | 40         |
+| Week 10-12      | Documentation, final polishing, mentor review                 | 35         |
+
+### Stretch Goals
+
+- Implement auto-update functionality
+- Create GUI installer for each platform
```

Comment:
Need to make sure it isn't only installable via a GUI, as there are many systems (e.g. HPC) that we target without a display